text/0000-cfg_attr-multiple-attrs.md

```diff
@@ -0,0 +1,177 @@
+- Feature Name: cfg_attr_multi
+- Start Date: 2018-09-10
+- RFC PR: 
+- Rust Issue: 
+
+# Summary
+[summary]: #summary
+
+Change `cfg_attr` to allow multiple attributes after the configuration
+predicate, instead of just one. When the configuration predicate is true,
+replace the attribute with all following attributes.
+
+# Motivation
+[motivation]: #motivation
+
+Simply put, ergonomics and intent. When you have multiple attributes you
+configure away behind the same predicate today, you need to duplicate the entire
+predicate. And then when you read code that does this, you have to check the
+entire predicates with each other to make sure they're the same. By allowing
+multiple attributes it removes that duplication and shows explicitly that the
+author wanted those attributes configured behind the same predicate.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The `cfg_attr` attribute takes a configuration predicate and then a list of
+attributes that will be in effect when the predicate is true.
+
+For an example of multiple attributes, say we want to have two attribute macros
+(`sparkles` and `crackles`), but only when `feature = "magic"` is enabled. We
+can write this as:
+
+```rust,igore
+#[cfg_attr(feature = "magic", sparkles, crackles)]
+fn bewitched() {}
+```
+
+When the feature flag is enabled, it expands to:
+
+```rust,ignore
+#[sparkles]
+#[crackles]
+fn bewitche() {}
+```
+
+The list of attributes may be empty, but will warn if the actual source code
+contains an empty list.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The next section replaces what's in the Conditional Compilation Chapter for the
+`cfg_attr` attribute. It explains both current and new behavior, mainly because
+the current reference material needs improvement.
+
+## `cfg_attr` Attribute
+
+The `cfg_attr` attribute conditionally includes attributes based on a
+configuration predicate. 
+
+It is written as `cfg_attr` followed by `(`, a comma separated metaitem
+sequence, and then `)` The metaitem sequence contains one or more metaitems.
+The first is a conditional predicate. The rest are metaitems that are also
+attributes. Trailing commas are permitted. The following list are all allowed:
+
+* `cfg_attr(predicate, attr)`
+* `cfg_attr(predicate, attr_1, attr_2)`
+* `cfg_attr(predicate, attr,)`
+* `cfg_attr(predicate, attr_1, attr_2,)`
+* `cfg_attr(predicate)`
```

Comment:
Nit: this is better prohibited for uniformity.

Imagine any other syntax for the predicate/attrs separator, e.g. `cfg_attr(predicate: attr1, attr2)`, then empty attribute list would look like `cfg_attr(predicate:)` with `cfg_attr(predicate)` being invalid, i.e. the first comma is not like the others.

Reply:
We usually go for zero-or-more instead of one-or-more in most places where we allow -or-more? so I'm not sure restricting this to one-or-more is the uniform thing to do here?

Reply:
I mean `cfg_attr(predicate, /* no attrs */)` - OK, `cfg_attr(predicate /* no attrs */)` - ERROR, i.e. zero-or-more exactly.

Reply:
Oh; I see now :)
I have no strong feelings here.

Are there any future compat hazards with allowing the omission of the `,` ?
I don't think macros benefit from omitting `,`.

Reply:
I think I agree here.  Allowing `cfg_attr(p)` is like allowing both leading _and_ trailing commas; we don't need it, and blocking it simplifies the grammar.

Reply:
@Havvy can you make the restriction based on @petrochenkov's and @scottmcm's comments?
We can always relax in the future if we feel so inclined but for now we can be more conservative here until we have a good justification to do something else.

Reply:
Done.